addons/controllers/calico/calicoconfig_utils.go

```diff
@@ -33,6 +40,56 @@ type config struct {
 	SkipCNIBinaries bool   `yaml:"skipCNIBinaries"`
 }
 
+// ClusterToCalicoConfig returns a list of Requests with CalicoConfig ObjectKey
+func (r *CalicoConfigReconciler) ClusterToCalicoConfig(o client.Object) []ctrl.Request {
```

Comment:
nit: maybe register this event handler with the `calicoconfig` controller like others?